File: android/tutorials/tutorial1/src/main/java/com/ubercab/presidio_training1/root/RootView.java
package com.ubercab.presidio_training1.root;

import android.content.Context;
import android.support.annotation.Nullable;
import android.util.AttributeSet;
import android.widget.FrameLayout;

/** Top level view for {@link RootBuilder.RootScope}. */
class RootView extends FrameLayout implements RootInteractor.RootPresenter {

  public RootView(Context context) {
    this(context, null);
  }

  public RootView(Context context, @Nullable AttributeSet attrs) {
    this(context, attrs, 0);
  }

  public RootView(Context context, @Nullable AttributeSet attrs, int defStyle) {
    super(context, attrs, defStyle);
  }
}

File: android/demos/coroutines/src/main/kotlin/com/uber/rib/root/RootView.kt
package com.uber.rib.root

import android.content.Context
import android.util.AttributeSet
import android.view.View

/**
 * Top level view for {@link RootBuilder.RootScope}.
 */
class RootView @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyle: Int = 0) : View(context, attrs, defStyle), RootInteractor.RootPresenter
